### PR TITLE
add infra other field

### PR DESCRIPTION
**Pull request by wu-xm**

### CODE REVIEW

File: controllers/infra/api/v1/infra_types.go [hunk 1]
```diff
@@ -160,7 +161,15 @@ type InfraSpec struct {
...
-
+	ID               types.UID   `json:"id,omitempty"`
+	Name             string      `json:"name,omitempty"`
+	RegionIDs        []string    `json:"regionIDs,omitempty"`
+	ZoneIDs          []string    `json:"zoneIDs,omitempty"`
+	SSH              v1bata1.SSH `json:"ssh,omitempty"`
+	Hosts            []Hosts     `json:"hosts,omitempty"`
+	CreateTime       string      `json:"createTime,omitempty"`
+	Connections      string      `json:"connections,omitempty"` //master0 ip:port
+	AvailabilityZone string      `json:"availabilityZone,omitempty"`
```
Comment (zzjin) [inline controllers/infra/api/v1/infra_types.go:172]: What is `AvailabilityZone` means?

Comment (wu-xm) [reply controllers/infra/api/v1/infra_types.go:172]: image

aws‘s Zone

File: controllers/infra/controllers/infra_controller.go [hunk 2]
```diff
@@ -67,7 +68,34 @@ func (r *InfraReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		r.recorder.Eventf(infra, "Error", "reconcile infra failed", "%v", err)
 		return ctrl.Result{}, err
 	}
-
+	Hosts, err := r.driver.GetInstances(infra)
```
[the remark this answers is not in the document]
Comment (wu-xm) [reply controllers/infra/controllers/infra_controller.go:71]: Yes, I see

File: controllers/infra/controllers/infra_controller.go [hunk 3]
```diff
@@ -61,13 +62,33 @@ func (r *InfraReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
...
 	}
-
+	//status
+	infra.Status.SSH = infra.Spec.SSH
+	var cstZone = time.FixedZone("CST", 8*3600)
+	infra.Status.CreateTime = time.Now().In(cstZone).Format("2006-01-02 15:04:05")
+	infra.Status.RegionIDs = infra.Spec.RegionIDs
+	infra.Status.AvailabilityZone = infra.Spec.AvailabilityZone
+	infra.Status.Connections = ""
+OuterLoop:
```
Comment (zzjin) [inline controllers/infra/controllers/infra_controller.go:77]: Do not use loop break.

Comment (zzjin) [inline controllers/infra/controllers/infra_controller.go:77]: Do not use loop break.

Comment (wu-xm) [reply controllers/infra/controllers/infra_controller.go:77]: ok

File: controllers/infra/drivers/aws/create_instance.go [hunk 4]
```diff
@@ -142,11 +142,11 @@ func (d Driver) createInstances(hosts *v1.Hosts, infra *v1.Infra) error {
...
-	if infra.Spec.SSH.PkName == "" {
-		if err := d.createKeyPair(infra); err != nil {
-			return err
-		}
-	}
+	//if infra.Spec.SSH.PkName == "" {
+	//	if err := d.CreateKeyPair(infra); err != nil {
+	//		return err
+	//	}
+	//}
```
Comment (zzjin) [inline controllers/infra/drivers/aws/create_instance.go:149]: If not used anymore, just delete this code block

File: controllers/infra/drivers/reconcile.go [hunk 5]
```diff
@@ -81,6 +81,14 @@ func (a *Applier) ReconcileHosts(current []v1.Hosts, infra *v1.Infra, driver Dri
 		d := desired[i]
 		cur := getHostsByIndex(d.Index, current)
 		eg.Go(func() error {
+			if cur == nil || d.Count > cur.Count {
+				if infra.Spec.SSH.PkName == "" {
+					if err := driver.CreateKeyPair(infra); err != nil {
+						return err
+					}
+				}
+				infra.Status.SSH = infra.Spec.SSH
```
Comment (zzjin) [inline controllers/infra/drivers/reconcile.go:90]: When the status is updated, do we need to call `status().Update()`?

File: controllers/infra/drivers/aws/create_instance.go (hunk 4 above)
Comment (wu-xm) [reply controllers/infra/drivers/aws/create_instance.go:149]: ok

File: controllers/infra/api/v1/infra_types.go [hunk 6]
```diff
@@ -160,7 +161,15 @@ type InfraSpec struct {
...
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
-
+	ID               types.UID   `json:"id,omitempty"`
+	Name             string      `json:"name,omitempty"`
+	RegionIDs        []string    `json:"regionIDs,omitempty"`
+	ZoneIDs          []string    `json:"zoneIDs,omitempty"`
+	SSH              v1bata1.SSH `json:"ssh,omitempty"`
+	Hosts            []Hosts     `json:"hosts,omitempty"`
+	CreateTime       string      `json:"createTime,omitempty"`
```
Comment (fanux) [inline controllers/infra/api/v1/infra_types.go:170]: Seems metadata already has the create timestap.

File: controllers/infra/api/v1/infra_types.go [hunk 7]
```diff
@@ -160,7 +161,15 @@ type InfraSpec struct {
 type InfraStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
-
+	ID               types.UID   `json:"id,omitempty"`
+	Name             string      `json:"name,omitempty"`
```
Comment (fanux) [inline controllers/infra/api/v1/infra_types.go:165]: Why not use metadate.Name?

File: controllers/infra/controllers/infra_controller.go [hunk 8]
```diff
@@ -68,13 +67,35 @@ func (r *InfraReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		return ctrl.Result{}, err
 	}
 
+	//status
+	var cstZone = time.FixedZone("CST", 8*3600)
+	infra.Status.CreateTime = time.Now().In(cstZone).Format("2006-01-02 15:04:05")
+	infra.Status.RegionIDs = infra.Spec.RegionIDs
```
Comment (fanux) [inline controllers/infra/controllers/infra_controller.go:73]: If the spec already has RegionID, Why does status need to add this field repeatedly

File: controllers/infra/controllers/infra_controller.go [hunk 9]
```diff
@@ -68,13 +67,35 @@ func (r *InfraReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
...
+	infra.Status.AvailabilityZone = infra.Spec.AvailabilityZone
+	infra.Status.Connections = ""
+	ConnectionsFlag := false
+	for _, v := range infra.Status.Hosts {
+		for _, role := range v.Roles {
+			if role == "master" && len(v.Metadata) > 0 {
+				infra.Status.Connections = v.Metadata[0].IP[1].IPValue + ":22"
```
Comment (fanux) [inline controllers/infra/controllers/infra_controller.go:80]: hardcode

File: controllers/infra/api/v1/infra_types.go (hunk 6 above)
Comment (wu-xm) [reply controllers/infra/api/v1/infra_types.go:170]: Yes, I'll change it

File: go.mod [hunk 10]
```diff
@@ -159,21 +159,53 @@ require (
 	github.com/prometheus/client_model v0.2.0 // indirect
 	github.com/prometheus/common v0.37.0 // indirect
 	github.com/prometheus/procfs v0.8.0 // indirect
-	github.com/rivo/uniseg v0.2.0 // indirect
```
Comment (zzjin) [inline go.mod:162]: Do not direct edit root directory module file.

**Review by zzjin — APPROVED** [review]

**Pull request merged**